scenario1/terminate_finished_jobs.py

```diff
@@ -0,0 +1,73 @@
+import re
+from kubernetes import client, config
+from kubernetes.client import Configuration
+from kubernetes.stream import stream
+
+import urllib3
+from urllib3.exceptions import InsecureRequestWarning
+
+urllib3.disable_warnings(InsecureRequestWarning)
+
+NAMESPACE = "blis"
+
+config.load_kube_config()
+
+c = Configuration().get_default_copy()
+c.verify_ssl = False
+
+v1_batch = client.BatchV1Api()
+v1_core = client.CoreV1Api()    
+
+
+try:
+    jobs = v1_batch.list_namespaced_job(namespace=NAMESPACE)
+    print(f"Active Jobs in namespace '{NAMESPACE}':")
+    if not jobs.items:
+        print("No jobs found in this namespace.")
+    else:
+        for job in jobs.items:
+            print(f"- {job.metadata.name}")
+            if job.status.active:
+                selector_labels = job.spec.selector.match_labels
+                    
+                label_selector_str = ",".join([f"{k}={v}" for k, v in selector_labels.items()])
+
+                # List pods in the namespace that match the job's label selector.
+                pods = v1_core.list_namespaced_pod(
+                    namespace=NAMESPACE,
+                    label_selector=label_selector_str
+                )
```

Comment:
I think you can just pass the selector label dict here. Why convert it to a string? https://docs.kr8s.org/en/stable/examples/listing_resources.html#list-pods-by-label-selector